tommath.h

```diff
@@ -270,38 +270,53 @@ mp_bool mp_isodd(const mp_int *a) MP_WUR;
 /* set to zero */
 void mp_zero(mp_int *a);
 
-/* set to a digit */
-void mp_set(mp_int *a, mp_digit b);
-
-/* set a double */
+/* get and set doubles */
+double mp_get_double(const mp_int *a) MP_WUR;
 mp_err mp_set_double(mp_int *a, double b) MP_WUR;
 
-/* set a 32-bit const */
-/* TODO void - never fails */ mp_err mp_set_int(mp_int *a, unsigned long b);
+/* get integer, set integer and init with integer (int32_t) */
+int32_t mp_get_int32(const mp_int *a) MP_WUR;
+void mp_set_int32(mp_int *a, int32_t b);
+mp_err mp_init_int32(mp_int *a, int32_t b) MP_WUR;
 
-/* set a platform dependent unsigned long value */
-/* TODO void - never fails */ mp_err mp_set_long(mp_int *a, unsigned long b);
+/* get integer, set integer and init with integer, behaves like two complement for negative numbers (uint32_t) */
+#define mp_get_uint32(a) ((uint32_t)mp_get_int32(a))
+void mp_set_uint32(mp_int *a, uint32_t b);
+mp_err mp_init_uint32(mp_int *a, uint32_t b) MP_WUR;
 
-/* set a platform dependent unsigned long long value */
-/* TODO void - never fails */ mp_err mp_set_long_long(mp_int *a, unsigned long long b);
+/* get integer, set integer and init with integer (int64_t) */
+int64_t mp_get_int64(const mp_int *a) MP_WUR;
+void mp_set_int64(mp_int *a, int64_t b);
 
-/* get a double */
-double mp_get_double(const mp_int *a) MP_WUR;
+/* get integer, set integer and init with integer, behaves like two complement for negative numbers (uint64_t) */
+#define mp_get_uint64(a) ((uint64_t)mp_get_int64(a))
+void mp_set_uint64(mp_int *a, uint64_t b);
 
-/* get a 32-bit value */
-unsigned long mp_get_int(const mp_int *a) MP_WUR;
+/* get magnitude */
+uint32_t mp_get_mag32(const mp_int *a) MP_WUR;
+uint64_t mp_get_mag64(const mp_int *a) MP_WUR;
 
-/* get a platform dependent unsigned long value */
-unsigned long mp_get_long(const mp_int *a) MP_WUR;
+/* get integer, set integer (long) */
+#define mp_get_slong(a)        (sizeof (long) == 8 ? (long)mp_get_int64(a) : (long)mp_get_int32(a))
+#define mp_set_llong(a, b)     (sizeof (long) == 8 ? mp_set_int64((a), (b)) : mp_set_int32((a), (int32_t)(b)))
```

Comment:
What's this? I couldn't find documentation
or just another typo?

Reply:
Oh shit. The issue is that get, set long is already taken. I should have stayed in stockholm instead of giving in to you :-P

Reply:
Before I had get_ul get_sl and then I changed it to get_ulong get_slong when I changed to get_sint etc

Reply:
Maybe we should stay with sint32 uint32 slong ulong etc, this way we are consistent and ugly in every dimension.

Reply:
Or we just revert to what I proposed before.

Reply:
it's too much, my head explodes, I have to go to bed, sorry.

just leave it for now, we'll find a good way :)

Reply:
I updated the latest commit now, it uses int32, uint32 etc but keeps slong for now. gn8